test/e2e/service.go

```diff
@@ -797,6 +805,29 @@ var _ = Describe("Services", func() {
 	})
 })
 
+// updateService fetches a service, calls the update function on it,
+// and then attempts to send the updated service. It retries up to 2
+// times in the face of timeouts and conflicts.
+func updateService(c *client.Client, namespace, serviceName string, update func(*api.Service)) (*api.Service, error) {
+	var service *api.Service
+	var err error
+	for i := 0; i < 3; i++ {
+		service, err = c.Services(namespace).Get(serviceName)
+		if err != nil {
+			return service, err
+		}
+
+		update(service)
+
+		service, err = c.Services(namespace).Update(service)
```

Comment:
You could also do an unsafe update, without specifying resourceVersion, or use Patch instead of Update.

That wouldn't solve the timeout problem, though.